libs/sdk-ui-dashboard/src/model/commands/base.ts

```diff
@@ -0,0 +1,36 @@
+// (C) 2021 GoodData Corporation
+/**
+ * All available command types.
+ *
+ * @internal
+ */
+export type DashboardCommandType =
+    | "GDC.DASHBOARD.CMD.LOAD"
+    | "GDC.DASHBOARD.CMD.SAVE"
+    | "GDC.DASHBOARD.CMD.SAVEAS"
+    | "GDC.DASHBOARD.CMD.RESET"
+    | "GDC.DASHBOARD.CMD.RENAME";
+
+/**
+ * Base type for all commands.
+ *
+ * Commands are dispatched using dispatcher obtained by useDashboardDispatcher(). All the dispatchers are 'contextual' -
+ * they target backend, workspace and dashboard in depending on the Dashboard component tree from which the dispatch
+ * is done.
+ *
+ * @internal
+ */
+export interface IDashboardCommand {
+    /**
+     * Command type. Always starts with "GDC.DASHBOARD.CMD"
+     */
+    readonly type: DashboardCommandType;
```

Comment:
It's a shame we do not have TypeScript 4.1 we could use template types for this.